### PR TITLE
Adding helpful methods for ContentPart, ContentType builders and Cont…

### PR DESCRIPTION
**Pull request by MikeAlhayek**
Adding
- ContentPartDefinitionBuilder
  -  `ContentPartDefinitionBuilder WithField<TField>(string fieldName)`
  - `ContentPartDefinitionBuilder WithField<TField>(string fieldName, Action<ContentPartFieldDefinitionBuilder> configuration)`
  - `Task<ContentPartDefinitionBuilder> WithFieldAsync<TField>(string fieldName, Func<ContentPartFieldDefinitionBuilder, Task> configuration)`

- ContentTypeDefinitionBuilder
  - `ContentTypeDefinitionBuilder WithPart<TPart>()`
  - `ContentTypeDefinitionBuilder WithPart<TPart>(string name)`
  - `ContentTypeDefinitionBuilder WithPart<TPart>(string name, Action<ContentTypePartDefinitionBuilder> configuration)`

- ContentItemExtensions
  - `bool TryGet<TPart>(this ContentItem contentItem, out TPart part)`
  - `bool TryGet<TPart>(this ContentItem contentItem, string name, out TPart part)`
  - `bool TryGet(this ContentItem contentItem, Type contentElementType, string name, out ContentElement part)`

- EntityExtensions
   - `bool TryGet<T>(this IEntity entity, out T aspect)`

### CODE REVIEW

File: src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs [hunk 1]
```diff
@@ -6,6 +6,39 @@ namespace OrchardCore.ContentManagement
 {
     public static class ContentItemExtensions
     {
+        public static bool TryGet<TPart>(this ContentItem contentItem, out TPart part) where TPart : ContentPart
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs:9]: I remember there's `As<Part>()` am I right?

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs:9]: Yes there is As<>, we also have Get<> and GetOrCreate<>

Comment (hishamco) [reply src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs:9]: > Yes there is As<>, we also have Get<>

What a difference between those?

Comment (MikeAlhayek) [reply src/OrchardCore/OrchardCore.ContentManagement.Abstractions/ContentItemExtensions.cs:16]: GetOrCreate: Gets a content part by its type or create a new one.
Both Get<> and As<>: Gets a content element by its name. They are the same thing

**Comment by MikeAlhayek** [comment]
@jtkech any feedback on these methods?

**Pull request merged**